backend/src/test/java/com/bakdata/conquery/integration/sql/SqlTestSpecParser.java

```diff
@@ -0,0 +1,139 @@
+package com.bakdata.conquery.integration.sql;
+
+import com.bakdata.conquery.apiv1.query.Query;
+import com.bakdata.conquery.integration.IntegrationTest;
+import com.bakdata.conquery.integration.common.RequiredTable;
+import com.bakdata.conquery.io.jackson.Jackson;
+import com.bakdata.conquery.io.jackson.View;
+import com.bakdata.conquery.models.datasets.Dataset;
+import com.bakdata.conquery.models.datasets.Table;
+import com.bakdata.conquery.models.datasets.concepts.Concept;
+import com.bakdata.conquery.models.exceptions.JSONException;
+import com.bakdata.conquery.models.identifiable.Identifiable;
+import com.bakdata.conquery.models.identifiable.ids.Id;
+import com.bakdata.conquery.models.identifiable.ids.IdUtil;
+import com.bakdata.conquery.models.worker.SingletonNamespaceCollection;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.deser.DeserializationProblemHandler;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import lombok.RequiredArgsConstructor;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class SqlTestSpecParser {
+
+    private final SqlStandaloneSupport support;
+
+    public SqlTestSpecParser() {
+        this.support = new SqlStandaloneSupport(new Dataset("test"));
+    }
+
+    /**
+     * Imports the required tables and concepts for mapping tables and concepts.
+     * MUST BE CALLED before trying to use parseQuery().
+     */
+    public void importRequiredData(SqlIntegrationTest testSpec) throws JSONException, IOException {
+        importTables(testSpec);
+        importConcepts(testSpec);
+    }
+
+    /**
+     * Tables and concepts MUST be imported upfront calling importRequiredData().
+     * Converts a raw JSON query into the corresponding CQElement representatives.
+     */
+    public Query parseQuery(JsonNode rawQuery) throws JSONException, IOException {
+        JavaType queryType = Jackson.MAPPER.getTypeFactory().constructParametricType(Query.class, new JavaType[0]);
```

Comment:
Warum brauchst du hier überhaupt das generic, mit CPSType sollte sich das Problem von selbst lösen.

Reply:
Das war fast alles 1:1 vom ConQueryTestSpecParser übernommen, da der nicht out of the box für uns funktioniert hat. Ich habe jetzt mit der Integration der Tests begonnen und damit können wir den auch benutzen. Soll ich deine Anmerkungen dann in eurer alten Klassen einarbeiten oder passt das so?